soc/espressif/esp32s3: undefined reference to 'log_const_soc' when CONFIG_PM=y

### Describe the bug

When building any application for ESP32-S3 with power management enabled (CONFIG_PM=y – which is the default), the linker fails with: undefined reference to log_const_soc

Target platform SoC: Espressif ESP32-S3

Work around is manually adding the following two lines to zephyr/soc/espressif/esp32s3/soc.c     "#include <zephyr/logging/log.h>
LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);"

Temporary alternative: disable power management with CONFIG_PM=n in prj.conf

### Regression

- [ ] This is a regression.

### Steps to reproduce

_No response_

### Relevant log output

```shell
/zephyr-sdk-0.17.4/xtensa-espressif_esp32s3_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32s3_zephyr-elf/12.2.0/../../../../xtensa-espressif_esp32s3_zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(power.c.obj):(.literal.pm_state_set+0xc): undefined reference to `log_const_soc'
collect2: error: ld re
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

Windows
 Zephyr SDK
Zephyr version: 4.3.99

### Additional Context

When building any application for ESP32-S3 with power management enabled (CONFIG_PM=y – which is the default), the linker fails with: undefined reference to log_const_soc

Target platform SoC: Espressif ESP32-S3

Work around is manually adding the following two lines to zephyr/soc/espressif/esp32s3/soc.c     "#include <zephyr/logging/log.h>
LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);"

Temporary alternative: disable power management with CONFIG_PM=n in prj.conf

--- a/soc/espressif/esp32/power.c
+++ b/soc/espressif/esp32/power.c
@@ -11,7 +11,7 @@
 #include <hal/rtc_io_hal.h>
 
 #include <zephyr/logging/log.h>
-LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+LOG_MODULE_REGISTER(soc_pm, CONFIG_SOC_LOG_LEVEL);
 
 static uint32_t intenable;
 

--- a/soc/espressif/esp32c2/power.c
+++ b/soc/espressif/esp32c2/power.c
@@ -9,7 +9,7 @@
 #include <esp_sleep.h>
 
 #include <zephyr/logging/log.h>
-LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+LOG_MODULE_REGISTER(soc_pm, CONFIG_SOC_LOG_LEVEL);
 
 /* Invoke Low Power/System Off specific Tasks */
 void pm_state_set(enum pm_state state, uint8_t substate_id)

--- a/soc/espressif/esp32c3/power.c
+++ b/soc/espressif/esp32c3/power.c
@@ -9,7 +9,7 @@
 #include <esp_sleep.h>
 
 #include <zephyr/logging/log.h>
-LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+LOG_MODULE_REGISTER(soc_pm, CONFIG_SOC_LOG_LEVEL);
 
 /* Invoke Low Power/System Off specific Tasks */
 void pm_state_set(enum pm_state state, uint8_t substate_id)

--- a/soc/espressif/esp32c6/power.c
+++ b/soc/espressif/esp32c6/power.c
@@ -9,7 +9,7 @@
 #include <esp_sleep.h>
 
 #include <zephyr/logging/log.h>
-LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+LOG_MODULE_REGISTER(soc_pm, CONFIG_SOC_LOG_LEVEL);
 
 /* Invoke Low Power/System Off specific Tasks */
 void pm_state_set(enum pm_state state, uint8_t substate_id)

--- a/soc/espressif/esp32h2/power.c
+++ b/soc/espressif/esp32h2/power.c
@@ -9,7 +9,7 @@
 #include <esp_sleep.h>
 
 #include <zephyr/logging/log.h>
-LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+LOG_MODULE_REGISTER(soc_pm, CONFIG_SOC_LOG_LEVEL);
 
 /* Invoke Low Power/System Off specific Tasks */
 void pm_state_set(enum pm_state state, uint8_t substate_id)

--- a/soc/espressif/esp32s2/power.c
+++ b/soc/espressif/esp32s2/power.c
@@ -9,7 +9,7 @@
 #include <esp_sleep.h>
 
 #include <zephyr/logging/log.h>
-LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+LOG_MODULE_REGISTER(soc_pm, CONFIG_SOC_LOG_LEVEL);
 
 static uint32_t intenable;
 

--- a/soc/espressif/esp32s3/power.c
+++ b/soc/espressif/esp32s3/power.c
@@ -9,7 +9,7 @@
 #include <esp_sleep.h>
 
 #include <zephyr/logging/log.h>
-LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+LOG_MODULE_REGISTER(soc_pm, CONFIG_SOC_LOG_LEVEL);
 
 static uint32_t intenable;
 

